### DISCUSSION TITLE
How to set up `cmp-git` with `blink.compat`?

### DISCUSSION THREAD

**Discussion by kunzaatko** [saghen/blink.cmp#551] [Q&A]
I am struggling to set up [`cmp-git`](https://github.com/petertriho/cmp-git) with the `blink.compat` plugin. I have this in my lazy config:
```lua
  dependencies = {
-- ...
    { 'saghen/blink.compat', lazy = true, config = true },
    { 'petertriho/cmp-git', dependencies = { 'nvim-lua/plenary.nvim', name = 'plenary' } },
  },
  opts = {
-- ...
    sources = {
      default = { 'lsp', 'path', 'luasnip', 'git', 'digraphs', 'buffer', 'spell' },
      providers = {
-- ...
        git = {
          name = 'git',
          module = 'blink.compat.source',
          opts = {
            filetypes = { 'gitcommit', 'NeogitCommitMessage', 'octo', 'gitrebase' },
            remotes = { 'upstream', 'origin', 'github' },
          },
        },
      },
    },
-- ...
    completion = {
      keyword = {
        regex = '[-_]\\|\\k\\|:\\|#\\|@\\|!', -- To enable the cmp-git trigger keys
      },
    },
-- ...
  }
```
But somehow the completion is not working. I believe that the rest of the configuration is not relevant to this. Is there something that I am missing?
Note that my other `blink.compat` sources are working alright with a analogous configuration and `cmp-git` is woking just fine with `cmp.nvim`

**Comment by PeterCardenas** [comment]
not sure if this is related, but i had to set it as `async = true` for the git provider so that other completions worked. i also put the lazypluginspec as separate and loading as `InsertEnter` with its own `require('cmp_git').setup` call to be a bit more agnostic of blink

**Comment by krovuxdev** [comment]
Have you trie this [git](https://github.com/Kaiser-Yang/blink-cmp-git) ? Does it you work for you?
